cli/update/update.go

```diff
@@ -56,9 +56,9 @@ func runUpdateCommand(cmd *cobra.Command, args []string) {
 	inst := instance.CreateInstanceAndRunFirstUpdate()
 	logrus.Info("Executing `arduino-cli update`")
 
-	err := commands.UpdateCoreLibrariesIndex(context.Background(), &rpc.UpdateCoreLibrariesIndexRequest{
-		Instance: inst,
-	}, output.ProgressBar())
+	err := commands.UpdateCoreLibrariesIndex(context.Background(), &rpc.UpdateCoreLibrariesIndexRequest{Instance: inst},
+		output.ProgressBar(),
+		output.PrintErrorFromDownloadResult(tr("Error updating index")))
```

Comment:
I think there is a problem with the progress bar:

```text
$ arduino-cli core update-index --additional-urls http://example.com/package_foo_index.json
Downloading index: package_index.tar.bz2 downloaded
 0 / ? [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------=]   0.00%Error updating index: Error downloading index 'http://example.com/package_foo_index.json': Server responded with: 404 Not Found
```

Note that that "Error updating index" was tacked on at the end of the progress bar.

I see that the progress bar disappears for the index download processes that are successful.

Reply:
great catch!